src/pages/iou/IOUCurrencySelection.js

```diff
@@ -74,6 +76,35 @@ function IOUCurrencySelection(props) {
     const selectedCurrencyCode = (lodashGet(props.route, 'params.currency', props.iou.currency) || CONST.CURRENCY.USD).toUpperCase();
     const iouType = lodashGet(props.route, 'params.iouType', CONST.IOU.MONEY_REQUEST_TYPE.REQUEST);
     const reportID = lodashGet(props.route, 'params.reportID', '');
+    const threadReportID = lodashGet(props.route, 'params.threadReportID', '');
+
+    // Decides whether to allow or disallow editing a money request
+    useEffect(() => {
+        // Do not dismiss the modal, when it is not the edit flow.
```

Comment:
It would be nice to centralize this kind of `canEdit` logic somewhere. I don't think it should exist in these views. It seems like it should either be in `IOUUtils.js` or `ReportUtils.js`. The same logic should be used for both:

- Preventing the modal from being accessed directly with the URL (which I think is what the intention is of the change in this file)
- Preventing the user from clicking on the action that would take them to this URL in the app (the other changes in this PR)

Having this logic spread out in multiple places makes it more difficult to maintain and follow.

Reply:
Agree, makes sense. It sounds like DRY. Will do it. Thank you!

Reply:
makes sense 👍 +1 to simplifying/abstracting away the useEffect logic further

Reply:
Done - 32e32c416d9e395fac592abe5f261fd0d84fe45a.